jena-arq/src/main/java/org/apache/jena/sparql/lang/ParserBase.java

```diff
@@ -190,6 +190,24 @@ protected Node stripSign(Node node) {
         return NodeFactory.createLiteral(lex, lang, dt) ;
     }
 
+    protected void checkString(String string, int line, int column) {
+        for ( int i = 0 ; i < string.length() ; i++ ) {
+            // Not "codePointAt" which does surrogate processing.
+            char ch = string.charAt(i);
+            // Check surrogate pairs are pairs.
+            if ( Character.isHighSurrogate(ch) ) {
+                i++;
+                if ( i == string.length() )
+                    throw new QueryParseException("Bad surrogate pair (end of string)", line, column);
+                char ch1 = string.charAt(i);
+                if ( ! Character.isLowSurrogate(ch1) ) {
+                    throw new QueryParseException("Bad surrogate pair (high surrogate not followed by low surrogate)", line, column);
+                }
+            } else if ( Character.isLowSurrogate(ch) )
+                throw new QueryParseException("Bad surrogate pair (low surrogate without high surrogate)", line, column);
+        }
+    }
```

Comment:
:+1: 

I checked for a method that does something like this on Apache Commons Text, and Apache Commons Lang, but there's none. I thought some user would have requested that, interesting.